### ISSUE TITLE
parquet-cli: patch "095c78fec3378189296d38fede1255b0a4d05fd4" does not work

### ISSUE THREAD

**Issue by kylemart** [Homebrew/homebrew-core#93135]
### `brew gist-logs <formula>` link OR `brew config` AND `brew doctor` output

```shell
kylemart@PowerBookPro ~ % brew config
HOMEBREW_VERSION: 3.3.10-51-g26941f0
ORIGIN: https://github.com/Homebrew/brew
HEAD: 26941f0bfd40bec21af461c1b0ed55267bf430cf
Last commit: 5 hours ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 2c560a62559502bfeae1c7e59e2512eaff96c453
Core tap last commit: 18 minutes ago
Core tap branch: master
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_CORE_GIT_REMOTE: https://github.com/Homebrew/homebrew-core
HOMEBREW_MAKE_JOBS: 10
Homebrew Ruby: 2.6.8 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: 10-core 64-bit arm_firestorm_icestorm
Clang: 13.0.0 build 1300
Git: 2.32.0 => /Applications/Xcode.app/Contents/Developer/usr/bin/git
Curl: 7.77.0 => /usr/bin/curl
macOS: 12.1-arm64
CLT: 13.2.0.0.1.1638488800
Xcode: 13.2.1
Rosetta 2: false


kylemart@PowerBookPro ~ % brew doctor
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: "config" scripts exist outside your system or Homebrew directories.
`./configure` scripts often look for *-config scripts to determine if
software packages are installed, and which additional flags to use when
compiling and linking.

Having additional scripts in your path can confuse software installed via
Homebrew if the config script overrides a system or Homebrew-provided
script of the same name. We found the following "config" scripts:
  /Users/kylemart/.pyenv/shims/python3.10-config
  /Users/kylemart/.pyenv/shims/python-config
  /Users/kylemart/.pyenv/shims/python3-config
```

### Verification

- [X] I ran `brew update` and am still able to reproduce my issue.
- [X] I have resolved all warnings from `brew doctor` and that did not fix my problem.

### What were you trying to do (and why)?

I'm using an M1 MacBook Pro. When trying to run `parquet` (parquet-cli installed via Brew) commands, I'm encountering exceptions related to snappy. A recent commit made to this repo should have patched the issue, but it seems to have not worked for me. Here is the PR: https://github.com/Homebrew/homebrew-core/pull/92069.

### What happened (include all command output)?

kylemart@PowerBookPro ~ % parquet head /Users/kylemart/Downloads/userdata1.parquet
Exception in thread "main" org.xerial.snappy.SnappyError: [FAILED_TO_LOAD_NATIVE_LIBRARY] no native library is found for os.name=Mac and os.arch=aarch64
	at org.xerial.snappy.SnappyLoader.findNativeLibrary(SnappyLoader.java:361)
	at org.xerial.snappy.SnappyLoader.loadNativeLibrary(SnappyLoader.java:195)
	at org.xerial.snappy.SnappyLoader.loadSnappyApi(SnappyLoader.java:167)
	at org.xerial.snappy.Snappy.init(Snappy.java:69)
	at org.xerial.snappy.Snappy.<clinit>(Snappy.java:46)
	at org.apache.parquet.hadoop.codec.SnappyDecompressor.decompress(SnappyDecompressor.java:62)
	at org.apache.parquet.hadoop.codec.NonBlockedDecompressorStream.read(NonBlockedDecompressorStream.java:51)
	at java.base/java.io.DataInputStream.readFully(DataInputStream.java:201)
	at java.base/java.io.DataInputStream.readFully(DataInputStream.java:172)
	at org.apache.parquet.bytes.BytesInput$StreamBytesInput.toByteArray(BytesInput.java:286)
	at org.apache.parquet.bytes.BytesInput.toByteBuffer(BytesInput.java:237)
	at org.apache.parquet.bytes.BytesInput.toInputStream(BytesInput.java:246)
	at org.apache.parquet.column.impl.ColumnReaderBase.readPageV1(ColumnReaderBase.java:680)
	at org.apache.parquet.column.impl.ColumnReaderBase.access$300(ColumnReaderBase.java:57)
	at org.apache.parquet.column.impl.ColumnReaderBase$3.visit(ColumnReaderBase.java:623)
	at org.apache.parquet.column.impl.ColumnReaderBase$3.visit(ColumnReaderBase.java:620)
	at org.apache.parquet.column.page.DataPageV1.accept(DataPageV1.java:120)
	at org.apache.parquet.column.impl.ColumnReaderBase.readPage(ColumnReaderBase.java:620)
	at org.apache.parquet.column.impl.ColumnReaderBase.checkRead(ColumnReaderBase.java:594)
	at org.apache.parquet.column.impl.ColumnReaderBase.consume(ColumnReaderBase.java:735)
	at org.apache.parquet.column.impl.ColumnReaderImpl.consume(ColumnReaderImpl.java:30)
	at org.apache.parquet.column.impl.ColumnReaderImpl.<init>(ColumnReaderImpl.java:47)
	at org.apache.parquet.column.impl.ColumnReadStoreImpl.getColumnReader(ColumnReadStoreImpl.java:82)
	at org.apache.parquet.io.RecordReaderImplementation.<init>(RecordReaderImplementation.java:271)
	at org.apache.parquet.io.MessageColumnIO$1.visit(MessageColumnIO.java:147)
	at org.apache.parquet.io.MessageColumnIO$1.visit(MessageColumnIO.java:109)
	at org.apache.parquet.filter2.compat.FilterCompat$NoOpFilter.accept(FilterCompat.java:177)
	at org.apache.parquet.io.MessageColumnIO.getRecordReader(MessageColumnIO.java:109)
	at org.apache.parquet.hadoop.InternalParquetRecordReader.checkRead(InternalParquetRecordReader.java:136)
	at org.apache.parquet.hadoop.InternalParquetRecordReader.nextKeyValue(InternalParquetRecordReader.java:225)
	at org.apache.parquet.hadoop.ParquetReader.read(ParquetReader.java:132)
	at org.apache.parquet.hadoop.ParquetReader.read(ParquetReader.java:136)
	at org.apache.parquet.cli.BaseCommand$1$1.advance(BaseCommand.java:363)
	at org.apache.parquet.cli.BaseCommand$1$1.<init>(BaseCommand.java:344)
	at org.apache.parquet.cli.BaseCommand$1.iterator(BaseCommand.java:342)
	at org.apache.parquet.cli.commands.CatCommand.run(CatCommand.java:73)
	at org.apache.parquet.cli.Main.run(Main.java:155)
	at org.apache.hadoop.util.ToolRunner.run(ToolRunner.java:76)
	at org.apache.parquet.cli.Main.main(Main.java:185)

### What did you expect to happen?

I expected the command to not throw an exception.

### Step-by-step reproduction instructions (by running `brew` commands)

```shell
Note: I performed the following steps on an M1 Mac.

Here are the steps I performed:
1. Updated Homebrew to the latest version.
2. Ran `brew install parquet-cli`
3. Downloaded this sample parquet file to my Downloads folder: https://github.com/Teradata/kylo/blob/master/samples/sample-data/parquet/userdata1.parquet.
4. Ran `parquet head /Users/kylemart/Downloads/userdata1.parquet`
```

**Comment by kylemart** [comment]
Hey @SMillerDev, @shanielh, please let me know if you're observing the same thing.

**Comment by gromgit** [comment]
Dunno much about Parquet, but I definitely can't replicate your error:
```
$ parquet head ~/Downloads/userdata1.parquet 
Argument error: Could not determine file format of /Users/aho/Downloads/userdata1.parquet.
```
Try reinstalling `parquet-cli` and all its dependents:
```
brew reinstall $(brew deps parquet-cli) parquet-cli
```

**Comment by kylemart** [comment]
Interesting, I tried reinstalling dependencies as you suggested, and for that file I'm getting this error:
```
kylemart@PowerBookPro downloads % parquet head userdata1.parquet
Argument error: INT96 is deprecated. As interim enable READ_INT96_AS_FIXED  flag to read as byte array.
```

But when running it against this [file](https://file.re/2022/01/14/part-r-00006-ddaee723-f3f6-4f25-a34b-3312172aa6d7snappy/?download&st=1642135404) (taken from pems_sorted, found [here](https://anson.ucdavis.edu/~clarkf/)), I'm getting the Snappy error:
```
kylemart@PowerBookPro downloads % brew reinstall $(brew deps parquet-cli) parquet-cli
==> Downloading https://ghcr.io/v2/homebrew/core/openjdk/manifests/17.0.1_1
Already downloaded: /Users/kylemart/Library/Caches/Homebrew/downloads/3b1b62cfad844ff54529068a2fb6ce6454afcb241b5052fc130c7d1fecfb9b33--openjdk-17.0.1_1.bottle_manifest.json
==> Downloading https://ghcr.io/v2/homebrew/core/openjdk/blobs/sha256:73913da0c2
Already downloaded: /Users/kylemart/Library/Caches/Homebrew/downloads/ade8faf91f707427e191233cbae3f1ce897fa1ca4ef94fc767ba32aab60ad22e--openjdk--17.0.1_1.arm64_monterey.bottle.tar.gz
==> Reinstalling openjdk
==> Pouring openjdk--17.0.1_1.arm64_monterey.bottle.tar.gz
🍺  /opt/homebrew/Cellar/openjdk/17.0.1_1: 640 files, 306.2MB
==> Running `brew cleanup openjdk`...
Disable this behaviour by setting HOMEBREW_NO_INSTALL_CLEANUP.
Hide these hints with HOMEBREW_NO_ENV_HINTS (see `man brew`).
==> Downloading https://ghcr.io/v2/homebrew/core/parquet-cli/manifests/1.12.0-1
Already downloaded: /Users/kylemart/Library/Caches/Homebrew/downloads/4f0fef42e5372b6af2b5856157f5f98913be49c051a473c3e84726e5fce7caf1--parquet-cli-1.12.0-1.bottle_manifest.json
==> Downloading https://ghcr.io/v2/homebrew/core/parquet-cli/blobs/sha256:d0534a
Already downloaded: /Users/kylemart/Library/Caches/Homebrew/downloads/f5775552c3c86e133ec97635ff5140ab97f112310dcff278d040218b95517eb8--parquet-cli--1.12.0.arm64_monterey.bottle.1.tar.gz
==> Reinstalling parquet-cli
==> Pouring parquet-cli--1.12.0.arm64_monterey.bottle.1.tar.gz
🍺  /opt/homebrew/Cellar/parquet-cli/1.12.0: 113 files, 78.4MB
==> Running `brew cleanup parquet-cli`...
kylemart@PowerBookPro downloads % parquet head userdata1.parquet
Argument error: INT96 is deprecated. As interim enable READ_INT96_AS_FIXED  flag to read as byte array.
kylemart@PowerBookPro downloads % parquet head pems_sorted/station=402263/part-r-00000-ddaee723-f3f6-4f25-a34b-3312172aa6d7.snappy.parquet
Exception in thread "main" org.xerial.snappy.SnappyError: [FAILED_TO_LOAD_NATIVE_LIBRARY] no native library is found for os.name=Mac and os.arch=aarch64
	at org.xerial.snappy.SnappyLoader.findNativeLibrary(SnappyLoader.java:361)
	at org.xerial.snappy.SnappyLoader.loadNativeLibrary(SnappyLoader.java:195)
	at org.xerial.snappy.SnappyLoader.loadSnappyApi(SnappyLoader.java:167)
	at org.xerial.snappy.Snappy.init(Snappy.java:69)
	at org.xerial.snappy.Snappy.<clinit>(Snappy.java:46)
	at org.apache.parquet.hadoop.codec.SnappyDecompressor.decompress(SnappyDecompressor.java:62)
	at org.apache.parquet.hadoop.codec.NonBlockedDecompressorStream.read(NonBlockedDecompressorStream.java:51)
	at java.base/java.io.DataInputStream.readFully(DataInputStream.java:201)
	at java.base/java.io.DataInputStream.readFully(DataInputStream.java:172)
	at org.apache.parquet.bytes.BytesInput$StreamBytesInput.toByteArray(BytesInput.java:286)
	at org.apache.parquet.bytes.BytesInput.toByteBuffer(BytesInput.java:237)
	at org.apache.parquet.bytes.BytesInput.toInputStream(BytesInput.java:246)
	at org.apache.parquet.column.impl.ColumnReaderBase.readPageV1(ColumnReaderBase.java:680)
	at org.apache.parquet.column.impl.ColumnReaderBase.access$300(ColumnReaderBase.java:57)
	at org.apache.parquet.column.impl.ColumnReaderBase$3.visit(ColumnReaderBase.java:623)
	at org.apache.parquet.column.impl.ColumnReaderBase$3.visit(ColumnReaderBase.java:620)
	at org.apache.parquet.column.page.DataPageV1.accept(DataPageV1.java:120)
	at org.apache.parquet.column.impl.ColumnReaderBase.readPage(ColumnReaderBase.java:620)
	at org.apache.parquet.column.impl.ColumnReaderBase.checkRead(ColumnReaderBase.java:594)
	at org.apache.parquet.column.impl.ColumnReaderBase.consume(ColumnReaderBase.java:735)
	at org.apache.parquet.column.impl.ColumnReaderImpl.consume(ColumnReaderImpl.java:30)
	at org.apache.parquet.column.impl.ColumnReaderImpl.<init>(ColumnReaderImpl.java:47)
	at org.apache.parquet.column.impl.ColumnReadStoreImpl.getColumnReader(ColumnReadStoreImpl.java:82)
	at org.apache.parquet.io.RecordReaderImplementation.<init>(RecordReaderImplementation.java:271)
	at org.apache.parquet.io.MessageColumnIO$1.visit(MessageColumnIO.java:147)
	at org.apache.parquet.io.MessageColumnIO$1.visit(MessageColumnIO.java:109)
	at org.apache.parquet.filter2.compat.FilterCompat$NoOpFilter.accept(FilterCompat.java:177)
	at org.apache.parquet.io.MessageColumnIO.getRecordReader(MessageColumnIO.java:109)
	at org.apache.parquet.hadoop.InternalParquetRecordReader.checkRead(InternalParquetRecordReader.java:136)
	at org.apache.parquet.hadoop.InternalParquetRecordReader.nextKeyValue(InternalParquetRecordReader.java:225)
	at org.apache.parquet.hadoop.ParquetReader.read(ParquetReader.java:132)
	at org.apache.parquet.hadoop.ParquetReader.read(ParquetReader.java:136)
	at org.apache.parquet.cli.BaseCommand$1$1.advance(BaseCommand.java:363)
	at org.apache.parquet.cli.BaseCommand$1$1.<init>(BaseCommand.java:344)
	at org.apache.parquet.cli.BaseCommand$1.iterator(BaseCommand.java:342)
	at org.apache.parquet.cli.commands.CatCommand.run(CatCommand.java:73)
	at org.apache.parquet.cli.Main.run(Main.java:155)
	at org.apache.hadoop.util.ToolRunner.run(ToolRunner.java:76)
	at org.apache.parquet.cli.Main.main(Main.java:185)
kylemart@PowerBookPro downloads %
```

**Comment by carlocab** [comment]
What happens when you try to run `parquet head userdata1.parquet`:

1. with the version of `parquet-cli` formula without the `snappy` patch?
2. with a non-Homebrew version of `parquet-cli`?

**Comment by kylemart** [comment]
If I run it without the patch, I get the same behavior as my last message — for both files, respectively. The only way to get parquet-cli without Homebrew is to build it manipulating. Using a manually build version of the parquet-cli (from the HEAD of the parquet-mr repo) results in expected behavior.

**Comment by kylemart** [comment]
Is anyone able to repro the issue with the second file I shared above? https://file.re/2022/01/14/part-r-00006-ddaee723-f3f6-4f25-a34b-3312172aa6d7snappy/?download&st=1642135404

**Comment by cho-m** [comment]
Just from installing bottle and looking at JARs, my guess is the patch wasn't sufficient to fix issue:
```console
❯ ls $(brew --prefix)/opt/parquet-cli/libexec/*snappy*
/opt/homebrew/opt/parquet-cli/libexec/snappy-java-1.1.8.jar
```

Snappy JAR is still on v1.1.8 while the upstream fix was to move use v1.1.8.3 (minimum v1.1.8.2 for M1 support)

As a test, you can try downloading a newer copy of JAR into libexec directory and see if it works.

---

I am not familiar with Maven, but maybe `copy-dependencies` is pulling from online (e.g. https://search.maven.org/artifact/org.apache.parquet/parquet-hadoop/1.12.0/jar) rather than the locally modified POM file. 

Someone who uses Maven may need to chime in on what is exactly happening and whether it is possible to override so that the `parquet-cli` can get the updated dependency version from `../parquet-hadoop/pom.xml`.

From local attempt, I see that running copy-dependencies from:
- `parquet-cli` directory downloads 1.1.8 version
- top-level directory downloads 1.1.8.3 version (though it will get every subpackages' dependencies)

**Comment by iMichka** [comment]
Just one comment, I also looked at `parquet-cli` because the test fails on Linux.
I found some related issues to the issue I see: https://issues.apache.org/jira/browse/PARQUET-2104

Looking at their bug tracker, and the activity on https://github.com/apache/parquet-mr, it looks like the project is slowly dying. Issues pile up, and commit activity is low. Maybe it's just stable enough now and activity is low, but I find this weird.
